### ISSUE TITLE
pre-commit hangs while stashing unstaged changes, requires 'enter' key to continue

### ISSUE THREAD

**Issue by mariaines** [pre-commit/pre-commit#3187]
### search you tried in the issue tracker

hangs, stash, enter key

### describe your issue

This started somewhat recently without any change to my pre-commit version or config, so I suspect it's something else on my machine and not pre-commit, but I'm struggling to debug and hoping anyone has pointers or has encountered this before.

During commit, when unstaged files are detected, pre-commit prints
```
[INFO] Stashing unstaged files to <...patchfile...>
```
and then hangs, seemingly doing nothing. 

In another terminal, while in this hung state, `git status` shows the unstaged changes are no longer in the index, so the stash seems to have succeeded. When I press the 'enter' key, pre-commit then proceeds to run hooks successfully, but then hangs again after restoring / stashing for the 2nd time (for commit-msg stage). Pressing 'enter' again lets it continue and complete successfully.

I am able to stash manually with `git stash` with no problems. 

Any ideas? TIA 🙏🏽 

### pre-commit --version

pre-commit 3.7.0

### .pre-commit-config.yaml

```yaml
default_install_hook_types: ['pre-commit', 'pre-push', 'commit-msg']
default_stages: ['pre-commit']

repos:
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.2.1
    hooks:
      - id: prettier
        types_or: [javascript, jsx, ts, tsx, css, scss, markdown, yaml]
        files: <redacted>
        args:
          [
            --config=<redacted>,
            --ignore-path=<redacted>,
          ]
  - repo: https://github.com/pre-commit/mirrors-eslint
    rev: v7.32.0
    hooks:
      - id: eslint
        alias: eslint-js
        name: lint js files
        files: <redacted>
        exclude: |
          (?x)^(
              <redacted>
          )$
        types: [file]
        args:
          - -c=<redacted>
        additional_dependencies:
          - eslint@7.32.0
          - eslint-config-react-app@6.0.0
          - eslint-plugin-cypress@2.8.1
          - eslint-plugin-known-imports@0.0.15-beta.2
          - eslint-plugin-react-hooks@4.2.0
          - eslint-plugin-react-native@3.11.0
          - "@typescript-eslint/eslint-plugin@4.22.0"
          - eslint-plugin-flowtype@5.2.0
          - eslint-plugin-import@2.22.0
          - eslint-plugin-jest@24.0.0
          - eslint-plugin-jsx-a11y@6.3.1
          - eslint-plugin-react@7.20.3
          - eslint-plugin-testing-library@3.9.0
          - confusing-browser-globals@1.0.11
          - react@16.14.0
          - typescript@4.4.3

  - repo: local
    hooks:
      - id: minify_pngs
        name: minify_pngs
        language: system
        entry: node <redacted>
        types: [png]
        files: <redacted>
      - id: no_gen_yield
        name: "Check gen-yield pattern is not used"
        types: [python]
        exclude: <redacted>
        entry: '@gen.coroutine|@coroutine|gen.Return|Return\('
        language: pygrep
      - id: compile_backend_strings
        name: "Compile backend localisation strings files"
        language: system
        entry: <redacted>
        files: <redacted>
        pass_filenames: false
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v3.4.0
    hooks:
      - id: no-commit-to-branch
        name: "Don't commit directly to master or release"
        args: ['--branch', 'master', '--branch', 'main', '--pattern', '^prod_push.*']
        stages: ['pre-commit', 'pre-push']
      - id: no-commit-to-branch
        name: "Branch must be <64 characters and alphanum with - or _"
        args: ['--pattern', '(.{63}.+|^[._-].*|.*[._-]$|.*[^a-zA-Z0-9-_].*)']
        stages: ['pre-commit', 'pre-push']
      - id: trailing-whitespace
        stages: ['pre-commit']
        exclude: |
          (?x)^(
              <redacted>
          )
      - id: end-of-file-fixer
        stages: ['pre-commit']
        exclude: |
          (?x)^(
              <redacted>
          )
      - id: check-yaml
        args: ["--unsafe"]
        exclude: <redacted>
      - id: check-added-large-files
      - id: check-merge-conflict
      - id: check-json
        files: <redacted>
        types: [file, json]
  - repo: https://github.com/asottile/pyupgrade
    rev: v3.3.2
    hooks:
      - id: pyupgrade
        args: ['--py37-plus']
        files: <redacted>
  - repo: https://github.com/psf/black
    rev: 22.3.0
    hooks:
      - id: black
```

**Comment by asottile** [comment]
can you ^C instead of enter and attach the log it generates?

I suspect you've got a rogue legacy hook maybe?

**Comment by mariaines** [comment]
Omg, it's magically fixed 🤦🏽‍♀️ 

Before filing this I upgraded to 3.7.0, problem still happened, then saw `~/.cache/pre-commit/pre-commit.log` was still referencing 3.3.0 so didn't include it because I thought it was irrelevant. I also _thought_ I did a `pre-commit clean` before filing, but maybe only tried that afterwards? And then also deleted pre-commit.log at some point. And now it's not happening anymore. Lol. 

Thanks for the quick help @asottile, will re-open and try ^C if the problem recurs!
